packages/core/src/utils.js

```diff
@@ -1265,3 +1265,13 @@ export function schemaRequiresTrueValue(schema) {
 
   return false;
 }
+
+export const descriptionId = id => `${id}__description`;
+
+export const helpId = id => `${id}__help`;
+
+export const errorsId = id => `${id}__errors`;
+
+export const ariaDescribedBy = id => ({
+  "aria-describedby": `${errorsId(id)} ${descriptionId(id)} ${helpId(id)}`,
```

Comment:
What if, for example, the help text is empty? Would it be correct in that case to still include helpId within aria-describedby, or should it be removed in that case?

Reply:
It is an issue. As far as I know, the specification does not say anything about missing element references. Still, https://validator.w3.org/ gives an error, if for example help element is missing.

VoiceOver reads only the existing elements without a problem when the input gets focus, as expected. I guess other tools work similarly. Order of the IDs also matter: I think it's best to have errors first. If field has errors, user gets that infomation first, without need to go through description and help every time.

I'd suggest we add all IDs, because it still improves accessibility.

Reply:
I see, thanks!